### PR TITLE
PLF-8389 : filter groups and spaces groups in suggesters.

### PR DESCRIPTION
**Pull request by waelghannem**
apply filtering in the end point used by suggesters in order to return only groups and spaces groups that should be displayed.

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java [hunk 1]
```diff
@@ -367,16 +376,18 @@ public Response suggestUsernames(@Context UriInfo uriInfo,
...
           String fullName = id.getProfile().getFullName();
           String userName = (String) id.getProfile().getProperty(Profile.USERNAME);
-          opt.setType("user");
-          opt.setValue(userName);
-          opt.setText(fullName);
-          opt.setAvatarUrl(id.getProfile() == null ? null : id.getProfile().getAvatarUrl());
-          excludedIdentityList.add(id);
-          opt.setOrder(4);
-          nameList.addOption(opt);
+          if (currentUser != null && currentUser.equals("root") || isSuperManager || isMemberOfAnyCurrentUserSpaces(currentUserSpaces, userName, spaceSrv)) {
```
Comment (vsellier) [inline component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: couldn't ``"root".equals(currentUser)`` remove some complexity of test ?

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java [hunk 2]
```diff
@@ -556,9 +567,8 @@ public Response suggestUsernames(@Context UriInfo uriInfo,
     return userInfos;
   }
 
-  private void addSpaceOrUserToList(List<Identity> identities, IdentityNameList options,
-                                   Space space, String typeOfRelation, int order) throws SpaceException {
-    SpaceService spaceSrv = getSpaceService(); 
+  public void addSpaceOrUserToList(List<Identity> identities, IdentityNameList options,
+                                    Space space, String typeOfRelation, int order, String currentUser, Boolean isSuperManager, List<Space> currentUserSpaces, SpaceService spaceSrv) throws SpaceException {
```
Comment (vsellier) [inline component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: Is it really useful to pass the spaceservice as a parameter of this method ?

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java (hunk 1 above)
Comment (waelghannem) [reply component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: yes, it helps. statement changed to "root".equals(currentUser).

File: component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java (hunk 2 above)
Comment (waelghannem) [reply component/service/src/main/java/org/exoplatform/social/service/rest/PeopleRestService.java]: spaceservice removed from addSpaceOrUserToList params list.

**Pull request merged**